mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -990,20 +990,20 @@ LogicalResult QuantizeLinearConverter::matchAndRewrite(
   Type biasType = outputType;
   if (bias) {
     biasType = getShapedElementTy(bias);
-    if (biasType.getIntOrFloatBitWidth() < 32) {
-      biasType = isa<IntegerType>(biasType) ? cast<Type>(rewriter.getI32Type())
-                                            : cast<Type>(rewriter.getF32Type());
-      bias = createCastOp(rewriter, loc, biasType, bias,
-                          op.getBias().getType().getElementType());
-    }
   }
-  Value asShort =
-      createCastOp(rewriter, loc, biasType, scaled,
-                   op.getScale().getType().getElementType(), origOutputType);
+  if ((bias || origOutputType != outputType) &&
```

Comment:
origOutputType != outputType shouldn't that be always true ?

From what i understand origOutputType is the type in migraphx dialect and outputType is the type in TOSA

Reply:
origOutputType is the original type (it could be signed, unsigned or signless int or float), outputType is the type after conversion (so, signless int or float). You can see how the type is converted in MIXRShapedToTensorConverter

Reply:
so they will be different only if it's a signed or unsigned int. Note that we are looking at the element type, not the type of the tensors. For example, !migraphx.shaped<1x4x5x5xf32, 100x1x20x4> will convert to tensor<100xf32>, but the element type is f32, it didn't change. However, !migraphx.shaped<1x4x5x5xui8, 100x1x20x4> will get converted to tensor<100xi8>.